From 0f7d56ffe630f621f5b0a7618c6171721c77190d Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Tue, 11 Jun 2019 18:34:39 +0800
Subject: [PATCH 29/91] glupload: fix direct_dma_upload perform glthread fail

if we are in direct dma mode, we only have one eglimage

Upstream-Status: Pending

(cherry picked from commit d01a0cfbfe1b5ec73c6b41489de406bb52cb2e61)
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 gst-libs/gst/gl/gstglupload.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/gl/gstglupload.c b/gst-libs/gst/gl/gstglupload.c
index 7df5ee0..218d0d3 100644
--- a/gst-libs/gst/gl/gstglupload.c
+++ b/gst-libs/gst/gl/gstglupload.c
@@ -1855,7 +1855,10 @@ _dma_buf_upload_perform_gl_thread (GstGLContext * context,
       GST_GL_MEMORY_ALLOCATOR (gst_allocator_find
       (GST_GL_MEMORY_EGL_ALLOCATOR_NAME));
 
-  n_mem = GST_VIDEO_INFO_N_PLANES (dmabuf->params->v_info);
+  if (dmabuf->direct)
+    n_mem = 1;
+  else
+    n_mem = GST_VIDEO_INFO_N_PLANES (dmabuf->params->v_info);
   for (i = 0; i < n_mem; i++) {
     if (!dmabuf->eglimage[i])
       return;
-- 
2.43.0

